server/implementation/src/main/java/io/smallrye/graphql/bootstrap/Config.java

```diff
@@ -75,8 +77,22 @@ default boolean shouldEmitEvents() {
         return isTracingEnabled() || isMetricsEnabled() || isValidationEnabled() || isEventsEnabled();
     }
 
-    default boolean logPayload() {
-        return false;
+    public enum logPayloadOption {
```

Comment:
I suggest moving this out of Config.java into it's own file and change to `LogPayloadOption` (capital L)

Reply:
In the same package do you want me to create a new file for it?

Reply:
That should be fine